File: src/riego/sensores/TemperaturaReaderTask.java
package riego.sensores;

public class TemperaturaReaderTask implements Runnable{
    private volatile double ultimaTemperatura = Double.NaN;
    private volatile boolean ejecutando = true;

    @Override
    public void run() {
        while (ejecutando) {
            try {
                ultimaTemperatura = leerSensor();
                System.out.printf("[Temperatura] %.2f °C%n", ultimaTemperatura);
                Thread.sleep(2000); // simula muestreo
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
                break;
            }
        }
    }

    private double leerSensor() {
        return -25 + Math.random() * 75; // entre -25 y 50 °C
    }

    public double getUltimaTemperatura() {
        return ultimaTemperatura;
    }

    public void detener() {
        ejecutando = false;
    }

}

File: src/riego/sensores/temperatura_reader_task.py
import threading, random, time
from src.patrones.observer.observable import Observable

class TemperaturaReaderTask(threading.Thread, Observable[float]):
    def __init__(self):
        threading.Thread.__init__(self, daemon=True)
        Observable.__init__(self)
        self._detenido = threading.Event()

    def run(self):
        while not self._detenido.is_set():
            temp = random.uniform(-25, 50)      # genera temperatura aleatoria
            self.notificar_observadores(temp)   # avisa a los observadores
            time.sleep(2.0)                     # espera 2 segundos

    def detener(self):
        self._detenido.set()
